airflow/dag_processing/processor.py

```diff
@@ -573,6 +573,7 @@ def _execute_task_callbacks(self, dagbag: DagBag, request: TaskCallbackRequest):
                 ti.start_date = simple_ti.start_date
                 ti.end_date = simple_ti.end_date
                 ti.try_number = simple_ti.try_number
+                ti.hostname = simple_ti.hostname
```

Comment:
probably calling `ti.refresh_from_db()` can do the trick instead of this?

Have you tested with this fix?

Reply:
Good idea, I will have a try.

Reply:
Resolved it and let the `ti.refresh_from_db()` action only runs in the failure_callback block due to performance concern.

Reply:
Actually, I think  #16804 is the better way to solve this problem. Sometimes DB resource is tight.

Reply:
@kaxil hello, could you please help to take a look?